src/mappings/_utils.test.ts

```diff
@@ -0,0 +1,211 @@
+            getInterBtcApiMock.mockImplementation(() => Promise.reject(Error("soz lol")));
+
+            await expect(getForeignAsset(fakeAsset.foreignAsset.id)).rejects.toThrow("soz lol");
+            expect(getInterBtcApiMock).toHaveBeenCalledTimes(1);
+            expect(libGetForeignAssetMock).not.toHaveBeenCalled();
+        });
+
+        it("should reject if assetRegistry.getForeignAssets rejects", async () => {
+            getInterBtcApiMock.mockImplementation(() => Promise.resolve(mockInterBtcApi));
+            libGetForeignAssetMock.mockImplementation(() => Promise.reject(Error("computer says no")));
+
+            await expect(getForeignAsset(fakeAsset.foreignAsset.id)).rejects.toThrow("computer says no");
+            expect(getInterBtcApiMock).toHaveBeenCalledTimes(1);
+            expect(libGetForeignAssetMock).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    describe("cacheForeignAssets", () => {
+        afterEach(() => {
+            libGetForeignAssetsMock.mockReset();
+            getInterBtcApiMock.mockReset();
+        });
+
+        it("should get all foreign assets from lib as expected", async () => {
+            const fakeAssets = [fakeAsset];
+            libGetForeignAssetsMock.mockImplementation(() => Promise.resolve(fakeAssets));
+            getInterBtcApiMock.mockImplementation(() => Promise.resolve(mockInterBtcApi));
+    
+            await cacheForeignAssets();
+            const actualCache = testHelpers.getForeignAssetsCache();
+    
+            expect(getInterBtcApiMock).toHaveBeenCalledTimes(1);
+            expect(libGetForeignAssetsMock).toHaveBeenCalledTimes(1);
+            expect(actualCache.has(fakeAsset.foreignAsset.id)).toBe(true);
+            expect(actualCache.get(fakeAsset.foreignAsset.id)).toBe(fakeAsset);
+        });
+
+        it("should reject if getInterBtcApi rejects", async () => {
+            getInterBtcApiMock.mockImplementation(() => Promise.reject(Error("nope")));
+
+            await expect(cacheForeignAssets()).rejects.toThrow("nope");
+            expect(getInterBtcApiMock).toHaveBeenCalledTimes(1);
+            expect(libGetForeignAssetsMock).not.toHaveBeenCalled();
+        });
+
+        it("should reject if assetRegistry.getForeignAssets rejects", async () => {
+            libGetForeignAssetsMock.mockImplementation(() => Promise.reject(Error("no assets for you")));
+            getInterBtcApiMock.mockImplementation(() => Promise.resolve(mockInterBtcApi));
+
+            await expect(cacheForeignAssets()).rejects.toThrow("no assets for you");
+            expect(getInterBtcApiMock).toHaveBeenCalledTimes(1);
+            expect(libGetForeignAssetsMock).toHaveBeenCalledTimes(1);
+        });
+
+        it("should set usdt asset id if found", async () => {
+            const oldUsdtAssetId = testHelpers.getUsdtAssetId() | 0;
+
+            const fakeUsdtAsset = {
+                foreignAsset: {
+                    id: oldUsdtAssetId + 13,
```

Comment:
Why +13, or is it just random increment?

Reply:
Yep, random. Just making sure it is not the same as some previous asset id in case it was set inside another test.